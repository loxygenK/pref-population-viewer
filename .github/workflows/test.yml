name: "Test"

on:
  pull_request:
    types:
      - opened
      - synchronize

jobs:
  check:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v3

      - uses: actions/cache@v2
        with:
          path: ~/.pnpm-store
          key: ${{ runner.os }}-pnpm_store-${{ hashFiles('**/pnpm-lock.yaml') }}
          restore-keys: |
            ${{ runner.os }}-pnpm_store-

      - uses: pnpm/action-setup@v2
        with:
          version: 6.32
          run_install: true

      - run: pnpm test

      - uses: actions/upload-artifact@v3
        if: failure()
        with:
          name: failed-test-recording
          path: |
            ./cypress/videos
            ./cypress/screenshots
