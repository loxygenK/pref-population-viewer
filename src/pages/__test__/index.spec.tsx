import Home from "~/pages/index.page";
import { renderStatefulComponent } from "~/test/renderStatefulComponent";

describe("index page", () => {
  it("renders correctly", async () => {
    const rendered = await renderStatefulComponent((root) => {
      root.render(<Home />);
    });

    expect(rendered).toMatchSnapshot();
  });
});
