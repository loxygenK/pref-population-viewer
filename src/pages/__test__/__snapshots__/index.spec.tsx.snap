// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index page renders correctly 1`] = `
<div
  className="page_wrapper"
>
  <article
    className="content"
  >
    <section
      className="pref_list"
    >
      <h2
        className="section_heading"
      >
        都道府県
      </h2>
      <ul
        className="checkbox_list"
      >
        <li>
          <input
            checked={false}
            id="check-pref-0"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="check-pref-0"
          >
            Pref #0
          </label>
        </li>
        <li>
          <input
            checked={false}
            id="check-pref-1"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="check-pref-1"
          >
            Pref #1
          </label>
        </li>
        <li>
          <input
            checked={false}
            id="check-pref-2"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="check-pref-2"
          >
            Pref #2
          </label>
        </li>
        <li>
          <input
            checked={false}
            id="check-pref-3"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="check-pref-3"
          >
            Pref #3
          </label>
        </li>
        <li>
          <input
            checked={false}
            id="check-pref-4"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="check-pref-4"
          >
            Pref #4
          </label>
        </li>
      </ul>
    </section>
    <section
      className="population_graph"
    >
      <h2
        className="section_heading"
      >
        人口数
      </h2>
      <div
        className="suggestion_area"
      >
        <span
          className="suggestion_text"
        >
          都道府県リストから表示する都道府県を選択してください
        </span>
      </div>
    </section>
  </article>
  <aside
    className="credits"
  >
    RESAS (地域経済分析システム) 提供のデータを加工して作成
  </aside>
</div>
`;
